tests/binning_test.py

```diff
@@ -231,6 +231,17 @@ def test_bin_integer_coord_by_fractional_stepsize_raises(dtype):
         table.bin(label=sc.scalar(0.5, unit='m'))
 
 
+def test_bin_with_automatic_bin_bounds_raises_if_no_events():
+    table = sc.data.table_xyz(0)
+    with pytest.raises(ValueError):
+        table.bin(x=4)
+
```

Comment:
Does this also cover _binning_ on the already _binned_ data as we used as an example in the issue?

Reply:
The test does not cover that case explicitly, it only checks that the automatic selection of binning edges fails when there are no entries in the selected data.
I don't think it matters if the table was binned before or not, does it?

We could add a test for that, something like:

```python
def test_bin_with_automatic_bin_bounds_raises_if_no_events():
    table = sc.data.table_xyz(2)
    binned = table.bin(y=sc.array(dims=['y'], values=[0.0, 1.2, 2.0], unit='m'))
    with pytest.raises(ValueError):
        binned['y', 1].bin(x=3)
```

Should that be a separate unit test or merged with the other test?

Reply:
Maybe it's not necessary anymore.
I asked because, before it was using `bins.size().sum()` and length of `values` depending if it's binned or not.